jwst/pipeline/tests/test_calwebb_image3.py

```diff
@@ -0,0 +1,118 @@
+import pytest
+import os
+import shutil
+from jwst.stpipe import Step
+from jwst.assign_wcs import AssignWcsStep
+from jwst.datamodels import ImageModel
+
+
+INPUT_FILE = "dummy_cal.fits"
+INPUT_FILE_2 = "dummy2_cal.fits"
+INPUT_ASN = "dummy_asn.json"
+OUTPUT_PRODUCT = "custom_name"
+LOGFILE = "run_asn.log"
+LOGCFG = "test_logs.cfg"
+
+
+@pytest.fixture(scope='module')
+def make_dummy_cal_file(tmp_cwd_module):
+    '''
+    Make and save a dummy cal file in the temporary working directory
+    Partially copied from test_calwebb_image2.py
+    '''
+
+    image = ImageModel((2048, 2048))
+    image.data[:, :] = 1
+    image.meta.instrument.name = 'NIRCAM'
+    image.meta.instrument.filter = 'F210M'
+    image.meta.instrument.pupil = 'CLEAR'
+    image.meta.exposure.type = 'NRC_IMAGE'
+    image.meta.observation.date = '2024-02-27'
+    image.meta.observation.time = '13:37:18.548'
+    image.meta.date = '2024-02-27T13:37:18.548'
+    image.meta.subarray.xstart = 1
+    image.meta.subarray.ystart = 1
+
+    image.meta.subarray.xsize = image.data.shape[-1]
+    image.meta.subarray.ysize = image.data.shape[-2]
+
+    image.meta.instrument.channel = 'SHORT'
+    image.meta.instrument.module = 'A'
+    image.meta.instrument.detector = 'NRCA1'
+
+    # bare minimum wcs info to get assign_wcs step to pass
+    image.meta.wcsinfo.crpix1 = 693.5
+    image.meta.wcsinfo.crpix2 = 512.5
+    image.meta.wcsinfo.v2_ref = -453.37849
+    image.meta.wcsinfo.v3_ref = -373.810549
+    image.meta.wcsinfo.roll_ref = 272.3237653262276
+    image.meta.wcsinfo.ra_ref = 80.54724018120017
+    image.meta.wcsinfo.dec_ref = -69.5081101864959
+
+    image = AssignWcsStep.call(image)
+
+    with image as dm:
+        dm.save(INPUT_FILE)
+
+
+@pytest.fixture(scope='module')
+def make_dummy_association(make_dummy_cal_file):
+    
+    shutil.copy(INPUT_FILE, INPUT_FILE_2)
+    os.system(f"asn_from_list -o {INPUT_ASN} --product-name {OUTPUT_PRODUCT} -r DMS_Level3_Base {INPUT_FILE} {INPUT_FILE_2}")
+
+
+@pytest.mark.parametrize("in_memory", [True, False])
+def test_run_image3_pipeline(make_dummy_association, in_memory):
+    '''
+    Two-product association passed in, run pipeline, skipping most steps
+    '''
+    # save warnings to logfile so can be checked later
+    logcfg_content = f"[*] \n \
```

Comment:
Can we find an alternative to this?

I have a suspicion that lines like this (which are in other files as well) contribute to our logging issues and random failures. I think what this is doing is changing every logger to level `INFO` and adding a handler that writes to this log file. I don't believe the level is reset after the call.

One option would be to open the expected output file and check the status of that file. Another would be to add a [LogWatcher](https://github.com/spacetelescope/jwst/blob/8efc26fdbda7936199d1a05cb57707d2c1cb0665/jwst/tests/helpers.py#L42). I vote for checking the output file.

Reply:
Thanks. I'll follow up with you on this separate from this PR. The logging is pretty insane and has many issues needing fixes.